homeassistant/components/mobile_app/webhook.py

```diff
@@ -77,6 +77,12 @@ def setup_registration(hass: HomeAssistantType, store: Store,
         _LOGGER.warning('Received invalid JSON from mobile_app')
         return empty_okay_response(status=HTTP_BAD_REQUEST)
 
+    if (req_data.get(ATTR_WEBHOOK_ENCRYPTED, False) is False and
+            registration[ATTR_SUPPORTS_ENCRYPTION]):
+        _LOGGER.warning("Refusing to accept unencrypted webhook from %s",
+                        registration[ATTR_DEVICE_NAME])
+        return empty_okay_response()
```

Comment:
We should not okay this. This is a `HTTP_BAD_REQUEST`.

Reply:
I'll fix this once the `error_response` helper added in #21850 is merged.